src/lib.rs

```diff
@@ -272,23 +286,24 @@ impl<T: Ord> OrderedCollection<T> {
     /// s.insert(89);
     /// s.insert(7);
     /// s.insert(12);
-    /// let a = OrderedCollection::from_sorted_iter(s.iter());
-    /// assert_eq!(a.find_gte(50), Some(&&89));
+    /// let a = OrderedCollection::from_sorted_iter(s.iter().copied());
+    /// assert_eq!(a.find_gte(50), Some(&89));
     /// ```
     ///
     pub fn from_sorted_iter<I>(iter: I) -> Self
     where
         I: IntoIterator<Item = T>,
-        I::IntoIter: ExactSizeIterator<Item = T>,
+        I::IntoIter: ExactSizeIterator,
     {
         let iter = iter.into_iter();
         let n = iter.len();
-        let mut context = (Vec::with_capacity(n), iter);
-        eytzinger_walk(&mut context, 0);
+        // vec with capacity n + 1 because we don't use index 0 and starts with 1
+        let mut context = (Vec::with_capacity(n + 1), iter);
+        eytzinger_walk(&mut context, 1);
         let (mut items, _) = context;
 
-        // it's now safe to set the length, since all `n` elements have been inserted.
-        unsafe { items.set_len(n) };
+        // here we assume all `n` elements from the iterator was inserted in items
```

Comment:
```suggestion
        // SAFETY: all `n` elements from the iterator was inserted in items.
        // [0] is uninitialized, but that's okay since the value type is `MaybeUninit`.
```